### ISSUE TITLE
Markdown preview: add svg text support

### ISSUE THREAD

**Issue by Angelk90** [zed-industries/zed#21319]
### Summary

Markdown renderer does not render `<text>` element inside SVGs. 

### Describe the bug / provide steps to reproduce it

As you can see from the images on zed when you preview a markdown but it contains an svg image that contains text, the text is not displayed on the svg.

Online:
<img width="642" alt="Screenshot 2024-11-29 alle 09 53 27" src="https://github.com/user-attachments/assets/e0c15e2a-32c8-4179-b685-05fcfe8f110a">

Zed:
<img width="140" alt="Screenshot 2024-11-29 alle 09 53 49" src="https://github.com/user-attachments/assets/d3cf729d-b972-4687-be08-f357ba57bd55">

Svg of reference:
https://github.com/zed-industries/zed/actions/workflows/ci.yml/badge.svg

### Environment

```
Zed: v0.164.0 (Zed Preview)
OS: macOS 14.1.1
Memory: 16 GiB
Architecture: x86_64
```

**Comment by jansol** [comment]
Specifically this seems to be a problem with the `<text>` element in SVG. I'd expect that if one were to open the file in inkscape, convert the text to paths and save the file, it would render fine.

**Comment by jansol** [comment]
yes, vscode is built on top of a web browser and web browsers have extensive support for svg features

**Comment by jansol** [comment]
No, that was a completely different topic

**Comment by notpeter** [comment]
Steps to reproduce:
1. Create an `file.svg` with text:
```md
<svg xmlns="http://www.w3.org/2000/svg" width="83" height="20">
    <path
        d="M0,3 C0,1.34 1.35,0 3.0,0 L40,0 L40,20 L3.0,20 C1.35,20 0,18.65 0,17 L0,3 Z"
    />
    <text fill="#00FF00"><tspan x="0" y="14">GR</tspan></text>
    <text fill="#FF0000"><tspan x="20" y="14">RE</tspan></text>
</svg>
```
2. Create a `file.md` with markdown:
```md
# hi

![file.svg](file.svg)
```
3. `zed file.md`
4. `cmd-k, v` (open markdown preview to the side)

<img width="783" alt="Screenshot 2024-12-16 at 2 12 00" src="https://github.com/user-attachments/assets/2dcbe9ef-eee2-4856-bc26-45f98fd923e2" />
